proto/src/main/proto/grpc.proto

```diff
@@ -218,25 +218,26 @@ message TradeInfo {
     string tradeId = 2;
     string shortId = 3;
     uint64 date = 4;
-    bool isCurrencyForTakerFeeBtc = 5;
-    uint64 txFeeAsLong = 6;
-    uint64 takerFeeAsLong = 7;
-    string takerFeeTxId = 8;
-    string depositTxId = 9;
-    string payoutTxId = 10;
-    uint64 tradeAmountAsLong = 11;
-    uint64 tradePrice = 12;
-    string tradingPeerNodeAddress = 13;
-    string state = 14;
-    string phase = 15;
-    string tradePeriodState = 16;
-    bool isDepositPublished = 17;
-    bool isDepositConfirmed = 18;
-    bool isFiatSent = 19;
-    bool isFiatReceived = 20;
-    bool isPayoutPublished = 21;
-    bool isWithdrawn = 22;
-    string contractAsJson = 23;
+    string role = 5;
+    bool isCurrencyForTakerFeeBtc = 6;
+    uint64 txFeeAsLong = 7;
+    uint64 takerFeeAsLong = 8;
+    string takerFeeTxId = 9;
+    string depositTxId = 10;
+    string payoutTxId = 11;
+    uint64 tradeAmountAsLong = 12;
+    uint64 tradePrice = 13;
+    string tradingPeerNodeAddress = 14;
+    string state = 15;
+    string phase = 16;
+    string tradePeriodState = 17;
+    bool isDepositPublished = 18;
+    bool isDepositConfirmed = 19;
+    bool isFiatSent = 20;
+    bool isFiatReceived = 21;
+    bool isPayoutPublished = 22;
+    bool isWithdrawn = 23;
+    string contractAsJson = 24;
```

Comment:
Would be easier to just add new fields at the end, but I guess there is some system to it.

Reply:
There is a system, but don't know what it's called yet.  I'm just trying to keep field 'types' grouped together as I add more -- protocol flags together, tx-ids together, price & amt together, json-contract always the last field,  ...

I thought about backward compatibility problems being caused by doing this, but I think it's safe to assume the CLI and server will always be in sync until a single server can support multiple clients on remote hosts.

Reply:
I think you should not assume that CLI and server is in sync. Basically, once we announce the API publicly the protocol is fixed and should only be added to and old fields deprecated but not changed.

Reply:
OK, I will stop re-ordering fields like this before the public announcement.